src/olympia/devhub/tests/test_views.py

```diff
@@ -3835,3 +3835,29 @@ def test_devhub_version_edit_page(self):
     def test_devhub_version_list_page(self):
         url = reverse('devhub.addons.versions', args=[self.addon.slug])
         self.assertNameAndNoXSS(url)
+
+
+class TestSwitchToUnlisted(UploadTest):
```

Comment:
The rest of the tests that deal with status changes are in `test_views_versions.py` so it would be better to have this there.  And/or modify some of the tests that test changes to unlisted status.